model/infinispan/src/main/java/org/keycloak/spi/infinispan/impl/remote/DefaultCacheRemoteConfigProviderFactory.java

```diff
@@ -0,0 +1,358 @@
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.configuration.cache.CacheMode;
+import org.jboss.logging.Logger;
+import org.keycloak.Config;
+import org.keycloak.config.CachingOptions;
+import org.keycloak.infinispan.util.InfinispanUtils;
+import org.keycloak.marshalling.Marshalling;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.provider.EnvironmentDependentProviderFactory;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.provider.ProviderConfigurationBuilder;
+import org.keycloak.spi.infinispan.CacheRemoteConfigProvider;
+import org.keycloak.spi.infinispan.CacheRemoteConfigProviderFactory;
+
+import javax.net.ssl.SSLContext;
+
+import static org.keycloak.connections.infinispan.InfinispanConnectionProvider.CLUSTERED_CACHE_NAMES;
+import static org.keycloak.connections.infinispan.InfinispanConnectionProvider.skipSessionsCacheIfRequired;
+import static org.keycloak.spi.infinispan.impl.Util.copyFromOption;
+import static org.wildfly.security.sasl.util.SaslMechanismInformation.Names.SCRAM_SHA_512;
+
+/**
+ * The default implementation for {@link CacheRemoteConfigProviderFactory} and {@link CacheRemoteConfigProvider}.
+ * <p>
+ * It is used when an external Infinispan cluster is enabled.
+ */
+public class DefaultCacheRemoteConfigProviderFactory implements CacheRemoteConfigProviderFactory, CacheRemoteConfigProvider, EnvironmentDependentProviderFactory {
+
+    private static final String PROVIDER_ID = "default";
+    private static final Logger logger = Logger.getLogger(MethodHandles.lookup().lookupClass());
+
+    // configuration
+    private static final String PROPERTIES_FILE = "propertiesFile";
+    private static final String CLIENT_INTELLIGENCE = "clientIntelligence";
+    private static final String HOSTNAME = "hostname";
+    private static final String PORT = "port";
+    private static final String TLS_ENABLED = "tlsEnabled";
+    private static final String TLS_SNI_HOSTNAME = "tlsSniHostname";
+    private static final String USERNAME = "username";
+    private static final String PASSWORD = "password";
+    private static final String CONNECTION_POOL_MAX_ACTIVE = "connectionPoolMaxActive";
+    private static final String CONNECTION_POOL_EXHAUSTED_ACTION = "connectionPoolExhaustedAction";
+    private static final String AUTH_REALM = "authRealm";
+    private static final String SASL_MECHANISM = "saslMechanism";
+
+    // configuration defaults
+    private static final String CLIENT_INTELLIGENCE_DEFAULT = ClientIntelligence.getDefault().name();
+    private static final int CONNECTION_POOL_MAX_ACTIVE_DEFAULT = 16;
+    private static final String CONNECTION_POOL_EXHAUSTED_ACTION_DEFAULT = ExhaustedAction.CREATE_NEW.name();
+    private static final String AUTH_REALM_DEFAULT = "default";
+    private static final String SASL_MECHANISM_DEFAULT = SCRAM_SHA_512;
```

Comment:
We can replace the above two properties with direct calls to `AuthenticationConfigurationBuilder.DEFAULT_REALM` and `DEFAULT_MECHANISM`.

Reply:
nice catch! Will do!

Reply:
I'll be keeping the `SASL_MECHANISM_DEFAULT` as Keycloak uses `SCRAM_SHA_512` and the Infinispan default is `SCRAM_SHA_256`.